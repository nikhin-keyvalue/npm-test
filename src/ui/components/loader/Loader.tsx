import Preact, { h } from 'preact';

import { LoaderTypes } from 'types';
import { ASSETS_PATH } from 'config';

import './style.scss';

const Loader: Preact.FunctionComponent<LoaderTypes> = ({ type }) => (
  <div className='zamp-loader-wrapper'>
    <img
      src={`${ASSETS_PATH}/public/gif/${type}-loader.gif`}
      className={`zamp-${type}-loader-gif-container`}
      alt='Loading...'
    />
  </div>
);

Loader.defaultProps = {
  type: 'default'
};

export default Loader;
