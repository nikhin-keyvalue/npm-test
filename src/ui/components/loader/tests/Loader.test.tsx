import { h } from 'preact';
import { configure, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { Adapter } from 'enzyme-adapter-preact-pure';

import Loader from '../Loader';

configure({ adapter: new Adapter() });

describe('Loader', () => {
  it('render components', () => {
    const wrapper = mount(<Loader />);
    const tree = shallowToJson(wrapper);

    expect(tree).toMatchSnapshot();
  });
});
