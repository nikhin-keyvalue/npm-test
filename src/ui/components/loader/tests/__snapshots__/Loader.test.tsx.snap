// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loader render components 1`] = `
<Loader
  type="default"
>
  <div
    className="zamp-loader-wrapper"
  >
    <img
      alt="Loading..."
      className="zamp-default-loader-gif-container"
      src="undefined/public/gif/default-loader.gif"
    />
  </div>
</Loader>
`;
